pkgs/top-level/all-packages.nix

```diff
@@ -5755,7 +5755,9 @@ with pkgs;
 
   joystickwake = callPackage ../tools/games/joystickwake { };
 
-  juce = darwin.apple_sdk_11_0.callPackage ../development/misc/juce { };
+  juce = callPackage ../development/misc/juce {
+    stdenv = if stdenv.isDarwin then overrideSDK stdenv "11.0" else stdenv;
```

Comment:
SDK-specific stdenvs don’t compose very well. Even if the adapter is available, people might think you have to use one of those instead of overriding the SDK in the stdenv you want to use. For example, if `overrideSDK stdenv` is the standard pattern, it’s obvious you should be able to do `overrideSDK llvmPackages_15.stdenv "11.0"` or `overrideSDK qt6Packages.stdenv "11.0"` or `overrideSDK python3.stdenv "11.0"` or even `overrideSDK python3.stdenv { darwinMinVersion = "10.13";}`. The latter is equivalent to what I had to do manually to fix pybind11 for clang 16.

Regarding the conditional, you’d need it even if there were a SDK-specific stdenv. I’ve intentionally chosen not to make the adapter platform-neutral. It’s for Darwin, so you need to conditionally use it only on Darwin. I’m pretty sure I’ve seen it mentioned that the current approach is a layering violation, and I agree. This PR fixes that by making the adapter work like other platform-specific stuff in nixpkgs.